### PR TITLE
construct tomcat server name base on install path and allow rename

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: src/Tomcat/TomcatController.ts [hunk 1]
```diff
@@ -84,10 +79,22 @@ export class TomcatController {
...
-
-        const tomcatServer: TomcatServer = new TomcatServer(serverName, tomcatInstallPath, storagePath);
+        const tomcatServer: TomcatServer = new TomcatServer(serverName, tomcatInstallPath, catalinaBasePath);
         this._tomcatModel.addServer(tomcatServer);
         vscode.commands.executeCommand('tomcat.tree.refresh');
+        return serverName;
+    }
+
+    public async renameServer(server: TomcatServer): Promise<void> {
+        const newName: string = await vscode.window.showInputBox({
```
Comment (andxu) [inline src/Tomcat/TomcatController.ts]: have a regex check against user input?

File: src/extension.ts [hunk 2]
```diff
@@ -26,7 +26,8 @@ export function activate(context: vscode.ExtensionContext): void {
     context.subscriptions.push(tomcatController);
     context.subscriptions.push(tomcatServerTree);
     context.subscriptions.push(vscode.window.registerTreeDataProvider('tomcatServerExplorer', tomcatServerTree));
-    context.subscriptions.push(vscode.commands.registerCommand('tomcat.tree.refresh', (element: TomcatServer) => tomcatServerTree.refresh(element)));
+    context.subscriptions.push(vscode.commands.registerCommand('tomcat.tree.refresh', (server: TomcatServer) => tomcatServerTree.refresh(server)));
```
Comment (andxu) [inline src/extension.ts]: tomcatServerTree.refresh is enough for replaceing " (server: TomcatServer) => tomcatServerTree.refresh(server))"

File: src/extension.ts [hunk 3]
```diff
@@ -26,7 +26,8 @@ export function activate(context: vscode.ExtensionContext): void {
...
+    context.subscriptions.push(vscode.commands.registerCommand('tomcat.server.rename', async (server: TomcatServer) => { tomcatController.renameServer(server); }));
```
Comment (andxu) [inline src/extension.ts]: tomcatController.renameServer

File: src/extension.ts (hunk 2 above)
Comment (SummerSun) [reply src/extension.ts]: why... It does not work..

File: src/extension.ts (hunk 3 above)
Comment (andxu) [reply src/extension.ts]: remove async, { and  }

**Pull request merged**